reactnd-project-myreads-starter/src/components/SearchBooks.js

```diff
@@ -0,0 +1,41 @@
+import React, {Component} from 'react'
+import * as BooksAPI from './../BooksAPI'
+import BookShelf from './BookShelf'
+
+class SearchPage extends Component {
+    state = {
+        books: [],
+    }
+
+    searchBook = (e) => {
+        BooksAPI.search(e.target.value).then((books) => {
+            this.setState({books})
+        });
+    }
+
+    render() {
+        return (
+            <div className="search-books">
+                <div className="search-books-bar">
+                    <a className="close-search" onClick={() => this.props.onCloseSearch(false)}>Close</a>
+                    <div className="search-books-input-wrapper">
+                        <input type="text" placeholder="Search by title or author" onChange={this.searchBook}/>
```

Comment:
this not a valid controlled component.

https://reactjs.org/docs/forms.html#controlled-components